### DISCUSSION TITLE
Shallow datadir directory trees when installing bottle

### DISCUSSION THREAD

**Discussion by sp1ritCS** [Homebrew/discussions#3207] [Tap maintenance and Homebrew development]
I have a Homebrew tap at [sp1ritCS/tap](https://github.com/sp1ritCS/homebrew-tap), that originally contained [notekit](https://github.com/sp1ritCS/homebrew-tap/blob/master/Formula/notekit.rb) and it's dependency [cLaTeXMath](https://github.com/sp1ritCS/homebrew-tap/blob/master/Formula/clatexmath.rb).

Because cLaTeXMath is going to be replaced with MicroTeX (same [repository](https://github.com/NanoMichael/MicroTeX), clm currently lives on `master`, while µTeX lives under the `openmath` branch for now), development has basically stalled. Due to changes in OS X (libc/libstdc++?), we have received complains from users in the notekit chatroom, because clatexmath failed building on their machines.

Thus I decided to package [sp1ritCS/tap/microtex](https://github.com/sp1ritCS/homebrew-tap/blob/master/Formula/microtex.rb) and [sp1ritCS/tap/notekit-sp1rit](https://github.com/sp1ritCS/homebrew-tap/blob/master/Formula/notekit-sp1rit.rb) (my personal fork, with a lot of experimental changes, and the branch that changes from clm to microtex). Building them on both a Ubuntu OCI container (with exposed WAYLAND_DISPLAY to the host) and on an OS X (Catalina I think) VM was working, so I published them and asked that particular user to try installing the notekit-sp1rit formula, to test if it was again possible to install notekit with a single command.

Since that user had the same OS X version as the GHA runner, it seems to have pulled in the binary bottle and installed it and for whatever reason, Homebrew decided to install the proper directory structure in `/usr/local/share/notekit/`, but without actually having placed the files there (it also didn't install two additional files into /usr/local/share/microtex/):
![A screenshot of a file manager showing the correct directory structure, but without them containing any files.](https://media.discordapp.net/attachments/834808258631696407/965795452433883197/screenshot.png)

I've then asked him to reinstall notekit-sp1rit, but this time with the `-s` flag, it worked as expected.*

I've just wanted to ask, if this is either intentional  behavior (but then, why does building it from source work) or a known bug - and if there is a workaround for this.

*well, not quite, --HEAD also needed to be specified, but that was probably my bug 😄 .

**Comment by SMillerDev** [comment]
Head means that it pulls the latest git version. It could very well be that the upstream build script was fixed in head but not in the bottled release.

**Reply by sp1ritCS** [reply]
@SMillerDev no, it was just that the revision did specify the latest commit inside my `unimath` branch, I just hadn't specified that in the Formula that (I just had the revision: commithash) and I assume brew does some shallow clone magic (i.e. only cloning master branch), that resulted in 
```
Failure while executing; `git checkout -f 5c9722e56898e3e00c18c7e919b3e6d44866b2c6 --` exited with 128. Here's the output:
fatal: reference is not a tree: 5c9722e56898e3e00c18c7e919b3e6d44866b2c6
```
(running that on my fully cloned repository worked without issue tho)
